runtime/src/chpl-file-utils.c

```diff
@@ -113,6 +117,78 @@ qioerr chpl_fs_exists(int* ret, const char* name) {
   return err;
 }
 
+int chpl_fs_get_cuid(void) {
+  return getuid();
+}
+
+/*
+  Quick implementation of strcpy() using qio_malloc(). If it already exists
+  I can easily replace this.
+*/
+char* copy_string(const char* input);
+char* copy_string(const char* input) {
+  char* result = NULL;
+  size_t len = strlen(input);
+
+  result = qio_malloc(len + 1);
+
+  if (result == NULL) { return NULL; }
+
+  result[len] = 0;
+  strncpy(result, input, len);
+
+  return result;
+}
+
+/* Common code shared by "chpl_fs_get_home()" and "chpl_fs_get_home_uid()". */
+static qioerr
+copy_passwd_login_dir(const char** outs, int* outf, struct passwd* pw) {
+  char* result_str = NULL;
+  int result_found = 0;
+
+  // If we've found the given user, attempt to copy their login dir.
+  if (pw != NULL) {
+    result_str = copy_string(pw->pw_dir);
+    if (result_str == NULL) { return qio_mkerror_errno(); }
+    result_found = 1;
+  }
+
+  *outs = (const char *) result_str;
+  *outf = result_found;
+
+  return 0;
+}
+
+qioerr
+chpl_fs_get_home(const char** outs, int* outf, const char* user) {
+  struct passwd* pw = NULL;
+
+  // Clear errno to distinguish an error from lookup failure.
+  errno = 0;
+
+  // This is neither reentrant nor threadsafe.
+  pw = getpwnam(user);
```

Comment:
... so why wouldn't you call getpwnam_r, the reentrant / thread safe version?

Reply:
I don't know that it's necessary in this context.  We'd be protecting against the user's home directory changing out from under us, but that seems like a super rare occurrence

Reply:
We don't access anything else in the result, so it shouldn't matter if any other details change

Reply:
I don't think that's what "not reentrant / threadsafe" means. It doesn't mean you might get different results on different calls in any sensible way. It means that you get Undefined Behavior if you call it from more than 1 thread at a time.

Anyway in this I don't see any significant downside to using getpwnam_r, even if the other one could work. Is there some reason you are imagining it to be worth ignoring the issue?

Reply:
Oh, that's more scary than I thought it was